### PR TITLE
feat: add hiphase

### PR DESCRIPTION
**Pull request by padraicc**
### This PR:

(If this is a release PR, no need to add following. Leave this part empty)
(Use the following lines to create a PR text body. Make sure to remove all non-relevant one after you're done)
(Repeat each field as many times as necessary)

Added: for new features.
Changed: for changes in existing functionality.
Deprecated: for soon-to-be removed features.
Removed: for now removed features.
Fixed: for any bug fixes.
Security: in case of vulnerabilities.

### Review and tests:
- [ ] Tests pass
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Code review
- [ ] `CHANGELOG.md` is updated
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: docs/softwares.md [hunk 1]
```diff
@@ -562,3 +562,24 @@ Normalization of vcf files. Left aligns INDELs and adds one reference allele. Af
 
 #RESOURCESSCHEMA__vt_normalize#
 
+
+## [hiphase](url_to_tool)
+Introduction to hiphase
```
Comment (jonca79) [inline docs/softwares.md:567]: Describe the program and put it in alphabetical order

File: workflow/rules/deepvariant.smk [hunk 2]
```diff
@@ -48,3 +48,21 @@ rule deepvariant:
...
+        vcf=temp("snv_indels/deepvariant/{sample}_{type}_{chr}.vcf.gz"),
+        gvcf=temp("snv_indels/deepvariant/{sample}_{type}_{chr}.g.vcf.gz")
+        if config.get("deepvariant", {}).get("output_gvcf", False)
+        else [],
+    params:
+        model_type=config.get("deepvariant", {}).get("model_type", "PACBIO"),
+        output_gvcf=lambda wildcards: get_gvcf_output(wildcards, "deepvariant"),
+        int_res=lambda wildcards: f"snv_indels/deepvariant/{wildcards.sample}_{wildcards.type}_{wildcards.chr}",
+        regions=lambda wildcards: f" --regions {wildcards.chr} ",
+        extra=config.get("deepvariant", {}).get("extra", ""),
```
Comment (jonca79) [inline workflow/rules/deepvariant.smk:68]: Should the parameters be the same as for deepvariant?

File: workflow/schemas/config.schema.yaml [hunk 3]
```diff
@@ -403,6 +402,23 @@ properties:
         type: string
         description: name or path to docker/singularity container
 
+
+
+# Entry added using Hydra-Genetics CLI - please move it into alphabetically sorted order
+  hiphase:
```
Comment (jonca79) [inline workflow/schemas/config.schema.yaml:408]: Alphabetical order plus remove comment

File: workflow/rules/hiphase.smk [hunk 4]
```diff
@@ -0,0 +1,62 @@
+__author__ = "Padraic Corcoran"
+__copyright__ = "Copyright 2024, Padraic Corcoran"
+__email__ = "padraic.corcoran@scilifelab.uu.se"
+__license__ = "GPL-3"
+
+
+rule hiphase:
```
Comment (jonca79) [inline workflow/rules/hiphase.smk:7]: Nice solution, like a wrapper but in the rule

File: workflow/schemas/resources.schema.yaml [hunk 5]
```diff
@@ -515,5 +515,28 @@ properties:
         type: string
         description: max execution time
 
+
+
+# Entry added using Hydra-Genetics CLI - please move it into alphabetically sorted order
```
Comment (jonca79) [inline workflow/schemas/resources.schema.yaml:520]: fix

File: workflow/schemas/rules.schema.yaml [hunk 6]
```diff
@@ -499,6 +499,48 @@ properties:
           vcf:
             type: string
             description: hard filtered vcf file with snv amd indels calls
+  
+  hiphase:
```
Comment (jonca79) [inline workflow/schemas/rules.schema.yaml:503]: alphabetical order

**Review by jonca79 — CHANGES_REQUESTED** [review]

File: workflow/rules/deepvariant.smk (hunk 2 above)
Comment (padraicc) [reply workflow/rules/deepvariant.smk:68]: The difference in this rule is the model_type which here is defaults to 'PACBIO' if the model_type is not set in the config

File: workflow/rules/hiphase.smk (hunk 4 above)
Comment (padraicc) [reply workflow/rules/hiphase.smk:7]: Thanks!

**Review by jonca79 — APPROVED** [review]

**Pull request merged**